### PR TITLE
Integration proxy

### PR DESCRIPTION
**Pull request by hav-noms**

### CODE REVIEW

File: contracts/IERC20.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
+pragma solidity 0.4.25;
+
+/**
+ * @title ERC20 interface
+ * @dev see https://github.com/ethereum/EIPs/issues/20
+ */
+interface IERC20 {
+    function totalSupply() external view returns (uint256);
```
[the remark this answers is not in the document]
Comment (hav-noms) [reply contracts/IERC20.sol]: @jacko125 yes but it then needs to be an abstract contract

File: contracts/IERC20.sol [hunk 2]
```diff
@@ -0,0 +1,34 @@
+pragma solidity 0.4.25;
```
[the remark this answers is not in the document]
Comment (hav-noms) [reply contracts/IERC20.sol]: Yeah thats a nice tidy up

File: contracts/Depot.sol [hunk 3]
```diff
@@ -33,9 +33,9 @@ pragma solidity 0.4.25;
 import "./SelfDestructible.sol";
 import "./Pausable.sol";
 import "./SafeDecimalMath.sol";
-import "./ISynthetix.sol";
-import "./ISynth.sol";
-import "./IFeePool.sol";
+import "./interfaces/ISynthetix.sol";
```
Comment (jacko125) [inline contracts/Depot.sol]: very nice!

File: contracts/Synthetix.sol [hunk 4]
```diff
@@ -994,7 +1003,22 @@ contract Synthetix is ExternStateToken {
         _;
     }
 
-    modifier onlySynth() {
+    // modifier onlySynth() {
```
Comment (jacko125) [inline contracts/Synthetix.sol]: to be deleted.

**Pull request merged**